### PR TITLE
Added instructions on how to load Pytorch models

### PR DESCRIPTION
**Pull request by Demba3**
Added instructions on how to load Pytorch models. 
I had tested it locally with two models here is one. Let me know if you need more information. 
```python
import pickle
import torch
import torch.nn as nn

class Model2(nn.Module):
    def __init__(self):
        super(Model2, self).__init__()
        self.fc1 = nn.Linear(15, 20)
        self.fc2 = nn.Linear(20, 10)
        self.fc3 = nn.Linear(10, 1)
        self.tanh = nn.Tanh()

    def forward(self, x):
        x = self.tanh(self.fc1(x))
        x = self.tanh(self.fc2(x))
        x = self.tanh(self.fc3(x))
        return x
# Initialize model
model = Model2()
new_state_dict = convert_state_dict(model.state_dict())

# Load the pickled data from a file
with open(path, "wb") as f:
    pickle.dump(new_state_dict, f)
#     new_state_dict = pickle.load(f)
    print(new_state_dict)
```
```rust 
let file = File::open(path).unwrap();
    let linear_state_dict: IndexMap<String, Vec<f64>> =
            de::from_reader(file, DeOptions::new()).unwrap();

    let mut linearModel = sequential!(
        Ix1,
        [
            Linear::new("fc1", 15, 20),
            Activation::Tanh,
            Linear::new("fc2", 20, 10),
            Activation::Tanh,
            Linear::new("fc3", 10, 1),
            Activation::Tanh
        ]
    );
    linearModel.load_state_dict(path);

    assert_eq!(linear_state_dict, linearModel.state_dict());
 ```

```python
import torch.nn as nn
import pickle

class ConvoModel(nn.Module):
    def __init__(self, in_channels, out_channels, kernel_size=(2,2), padding=(1,2), pool_size=(2,2), pool_stride=(2,1)):
        super(ConvoModel, self).__init__()
        self.conv2d = nn.Conv2d(in_channels, out_channels, kernel_size=kernel_size, stride=(1,1), padding=padding, bias=True)
        self.avgpool = nn.AvgPool2d(kernel_size=pool_size, stride=pool_stride, padding=(0,0), ceil_mode=False)
        self.sigmoid = nn.Sigmoid()

    def forward(self, x):
        x = self.conv2d(x)
        x = self.avgpool(x)
        x = self.sigmoid(x)
        return x

convoModel = ConvoModel(in_channels=3, out_channels=1)
new_state_dict = convert_state_dict(convoModel.state_dict())

# Load the pickled data from a file
with open(path, "wb") as f:
    pickle.dump(new_state_dict, f)
#     new_state_dict = pickle.load(f)
    print(new_state_dict)
```

``` rust
    let file = File::open(path).unwrap();
    let convo_state_dict: IndexMap<String, Vec<f64>> =
            de::from_reader(file, DeOptions::new()).unwrap();
    
    
    let name = "conv2d";
    let padding = (1, 2);
    let (in_channels, out_channels) = (3, 1);

    let pool_size = (2, 2);
    let pool_stride = (2, 1);

    let mut convoModel = sequential!(
        Ix3,
        [
            Conv2D::new(
                name,
                in_channels,
                out_channels,
                padding,
                Filter::new((2, 2), (1, 1), (1, 1)),
            ),
            AvgPool::new(pool_size, pool_stride, (0, 0), (1, 1)),
            Activation::Sigmoid
        ]
    );

    convoModel.load_state_dict(path);

    assert_eq!(convo_state_dict, convoModel.state_dict());
```

### CODE REVIEW

**Review by LazaroHurtado — CHANGES_REQUESTED** [review]
Thank you for taking on this task, just two small recommendations:
- It seems like the tests you mentioned in the PR description can be an integration test and it would be nice if you can include them!
- Can you elaborate some more as to why this python conversion is needed in the first place, this would help new developers

**Comment by Demba3** [comment]
@LazaroHurtado
- I will add it as an integration test if it is not an issue to add the extra pickle files. Plus, is the test really necessary considering we already have tests for load model function? Unless you break the load model function this still works as long as you saved the python model following the instructions we specified. 

- will add more context as to why conversion is needed.

**Comment by LazaroHurtado** [comment]
@Demba3 To me, adding the tests signifies that the python method we provided is valid for deserialization of Pytorch to micrograd models. So if some model is not being properly loaded we can expect it to most likely be user error on the Python side

**Comment by Demba3** [comment]
@LazaroHurtado Makes sense will start working on it. Just to clarify, I will use the python function to deserialize the Pytorch models, then use those files to write the tests for micrograd right?

**Review by LazaroHurtado — CHANGES_REQUESTED** [review]
LGTM, just two small requests:
- Can you move the binary model state dict files to their respective test folder, in this case, it would be "tests/layers". Also, we have the files themselves to be in all lowercase, a nitpick I know.
- Can you update @jrego43's Convolutional model's parameter order just as some small cleanup?

File: README.md [hunk 1]
```diff
@@ -32,3 +32,28 @@ Any type of contribution is welcome as long as it adds value! i.e
...
+We need the convert_state_dict() function to convert PyTorch tensors to lists because micrograd_rs can't unpickle PyTorch tensors. This conversion will allow micrograd_rs to load pytorch models without any issues.
+```python
+# we need this import to serialize the model in a compatible format 
+import pickle
+
+# changes PyTorch generated state dict to micrograd state dict
+def convert_state_dict(state_dict):
+    new_state_dict = {}
+    for name, tensor in state_dict.items():
+        new_state_dict[name] = [float(value) for value in tensor.flatten()]
```
Comment (LazaroHurtado) [inline README.md:46]:
```suggestion
        new_state_dict[name] = tensor.float().flatten().tolist()
```

**Review by LazaroHurtado — CHANGES_REQUESTED** [review]
LGTM, just one small suggestion for the python conversion method

**Review by LazaroHurtado — APPROVED** [review]
Good work!

**Review by jrego43 — APPROVED** [review]
Good stuff, LGTM.

**Pull request merged**